examples/policies/timed-mint-policy/timed-mint-policy-v1.pact

```diff
@@ -0,0 +1,134 @@
+(namespace (read-msg 'ns))
+
+(module timed-mint-policy-v1 GOVERNANCE
+
+  @doc "Policy for timed-mint tokens with royalty and quoted sale in coin."
+
+  (defconst ADMIN-KS:string "marmalade-v2.marmalade-contract-admin")
+
+  (defcap GOVERNANCE ()
+    (enforce-guard ADMIN-KS))
+
+  (implements kip.token-policy-v2)
+  (use kip.token-policy-v2 [token-info])
+  (use marmalade-v2.policy-manager)
+
+  (defschema timed-mint-schema
+    max-supply:decimal
+    mint-start-time:time
+    mint-end-time:time
+  )
+
+  (deftable timed-mint:{timed-mint-schema})
+
+  (defconst TIMED-MINT-SPEC:string "timed_mint_spec")
+
+  (defun validate-specs (spec:object{timed-mint-schema})
+    @doc "Enforce that the time windows and max supply conforms \
+    \to the contract requirements."
+
+    (let* ( (max-supply:decimal (at 'max-supply spec))
+            (mint-start-time:time (at 'mint-start-time spec))
+            (mint-end-time:time (at 'mint-end-time spec)) )
+
+      (enforce (>= max-supply 0.0) "Max supply must be non-negative")
+      (enforce (>= mint-end-time mint-start-time) "Mint end time must be after mint start time")
+      (enforce (>= mint-start-time (at 'block-time (chain-data))) "Mint start time must be in the future") ))
+
+   (defun enforce-init:bool
+     ( token:object{token-info}
+     )
+     (require-capability (INIT-CALL (at "id" token) (at "precision" token) (at "uri" token) timed-mint-policy-v1))
+     (let* ( (spec:object{timed-mint-schema} (read-msg TIMED-MINT-SPEC))
+             (max-supply:decimal (at 'max-supply spec))
+             (mint-start-time:time (at 'mint-start-time spec))
+             (mint-end-time:time (at 'mint-end-time spec))
+           )
+       (validate-specs spec)
+       (insert timed-mint (at 'id token)
+         { 'max-supply: max-supply
+         , 'mint-start-time:mint-start-time
+         , 'mint-end-time:mint-end-time
+         }))
+     true
+   )
+
+  (defun enforce-mint:bool
+    ( token:object{token-info}
+      account:string
+      guard:guard
+      amount:decimal
+    )
+    (require-capability (MINT-CALL (at "id" token) account amount timed-mint-policy-v1))
+    (let* ( (account-bal:decimal (try 0.0 (at 'balance (marmalade-v2.ledger.details (at 'id token) account))))
+            (total-supply:decimal (try 0.0 (marmalade-v2.ledger.total-supply (at 'id token))))
+            (timed-mint:object{timed-mint-schema} (get-timed-mint token))
+            (max-supply:decimal (at 'max-supply timed-mint))
+            (mint-start-time:time (at 'mint-start-time timed-mint))
+            (mint-end-time:time (at 'mint-end-time timed-mint)) )
+      (enforce (= account-bal 0.0) "Account has already minted")
+      (enforce (>= (at 'block-time (chain-data))  mint-start-time) "Mint has not started yet")
```

Comment:
We can replace `(at 'block-time (chain-data))` to `(curr-time)` function in `util-v1`, which returns the block time in timestamps.